subsys/dfu/boot/mcuboot.c

```diff
@@ -75,10 +75,14 @@ struct mcuboot_v1_raw_header {
 /* FLASH_AREA_ID() values used below are auto-generated by DT */
 #ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
 #define FLASH_AREA_IMAGE_PRIMARY FLASH_AREA_ID(image_0_nonsecure)
-#define FLASH_AREA_IMAGE_SECONDARY FLASH_AREA_ID(image_1_nonsecure)
+#define UPLOAD_FLASH_AREA_ID FLASH_AREA_ID(image_1_nonsecure)
 #else
 #define FLASH_AREA_IMAGE_PRIMARY FLASH_AREA_ID(image_0)
-#define FLASH_AREA_IMAGE_SECONDARY FLASH_AREA_ID(image_1)
+#if FLASH_AREA_LABEL_EXISTS(image_1)
+	#define UPLOAD_FLASH_AREA_ID FLASH_AREA_ID(image_1)
+#else
+	#define UPLOAD_FLASH_AREA_ID FLASH_AREA_ID(image_0)
+#endif
```

Comment:
I think it would be better to leave `FLASH_AREA_IMAGE_SECONDARY` defined (conditionally) and use *it* when doing swap state checks, and use `UPLOAD_FLASH_AREA_ID` only for operations where sending data to a partition is involved.

Reply:
I agree fully